open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/retrieval/flow/VRPConsentRetrievalUtilTest.java

```diff
@@ -0,0 +1,432 @@
+
+        PowerMockito.mockStatic(OpenBankingConfigParser.class);
+        PowerMockito.when(OpenBankingConfigParser.getInstance()).thenReturn(openBankingConfigParse);
+
+        Mockito.doReturn(ConsentExtensionConstants.PAYMENTS).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(paymentReceipt).when(consentResourceMock).getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.CREATED_TIME).when(consentResourceMock)
+                .getCreatedTime();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+
+        JSONArray paymentConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+
+        Assert.assertNotNull(paymentConsentData);
+    }
+
+    @Test
+    public void testGetConsentDataSetForCOF() {
+
+        Mockito.doReturn(ConsentExtensionConstants.FUNDSCONFIRMATIONS).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.COF_RECEIPT).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+
+        JSONArray cofConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+
+        Assert.assertNotNull(cofConsentData);
+    }
+
+    @Test
+    public void testGetConsentDataSetForVRPData() {
+
+        Mockito.doReturn(ConsentExtensionConstants.VRP).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.VRP_INITIATION).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+            JSONArray vrpConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+            Assert.assertNotNull(vrpConsentData);
+    }
+
+    @Test
+    public void testAppendDummyAccountID() {
+
+        JSONArray result = ConsentRetrievalUtil.appendDummyAccountID();
+
+        Assert.assertNotNull(result);
+
+        assertEquals(result.size(), 2);
+
+        JSONObject accountOne = (JSONObject) result.get(0);
+        assertEquals(accountOne.get("account_id"), "12345");
+        assertEquals(accountOne.get("display_name"), "Salary Saver Account");
+
+        JSONObject accountTwo = (JSONObject) result.get(1);
+        assertEquals(accountTwo.get("account_id"), "67890");
+        assertEquals(accountTwo.get("display_name"), "Max Bonus Account");
+    }
+
+    @Test(expectedExceptions = ConsentException.class)
+    public void testConsentDataWithInvalidJson() {
+        String invalidJsonString = "Invalid JSON String";
+        Mockito.when(consentResourceMock.getReceipt()).thenReturn(invalidJsonString);
+
+        try {
+            JSONArray consentDataJSON = ConsentRetrievalUtil.getConsentData(consentResourceMock);
```

Comment:
Here since we are testing negative case, we should not use try catch and catch the expected exception. If we want to check the error msg then we can remove the `(expectedExceptions = ConsentException.class)` from annotation and check the error msg.

Reply:
Fixed in [460b4c0]

Reply:
Fixed in [460b4c0]